File: frontend/src/api/evaluation.ts
import { apiClient } from "./client";
import type { EvaluationAssignmentPage, EvaluationSummary } from "../types/evaluation";

export async function getEvaluationSummary() {
  const { data } = await apiClient.get<EvaluationSummary>("/evaluation/summary");
  return data;
}

export async function getEvaluationAssignments(page = 1, search = "", action = "", strategy = "") {
  const { data } = await apiClient.get<EvaluationAssignmentPage>("/evaluation/assignments", { params: { page, page_size: 15, search: search || undefined, action: action || undefined, strategy: strategy || undefined } });
  return data;
}

File: backend/app/api/evaluation.py
from fastapi import APIRouter, Query

from app.schemas.evaluation import EvaluationAssignmentPage, EvaluationSummary
from app.services.evaluation_service import get_evaluation_assignments, get_evaluation_summary


router = APIRouter(prefix="/evaluation", tags=["evaluation"])


@router.get("/summary", response_model=EvaluationSummary)
def evaluation_summary() -> EvaluationSummary:
    return get_evaluation_summary()


@router.get("/assignments", response_model=EvaluationAssignmentPage)
def evaluation_assignments(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    search: str | None = None,
    action: str | None = None,
    strategy: str | None = None,
) -> EvaluationAssignmentPage:
    return get_evaluation_assignments(page, page_size, search, action, strategy)

